src/Microsoft.ML.PCA/PcaTrainer.cs

```diff
@@ -266,6 +307,27 @@ private static void PostProcess(VBuffer<Float>[] y, Float[] sigma, Float[] z, in
             }
         }
 
+        protected override SchemaShape.Column[] GetOutputColumnsCore(SchemaShape inputSchema)
+        {
+             return new[]
+            {
+                new SchemaShape.Column(DefaultColumnNames.Score,
+                        SchemaShape.Column.VectorKind.Scalar,
+                        NumberType.R4,
+                        false,
+                        new SchemaShape(MetadataUtils.GetTrainerOutputMetadata())),
+
+                new SchemaShape.Column(DefaultColumnNames.PredictedLabel,
+                        SchemaShape.Column.VectorKind.Scalar,
+                        BoolType.Instance,
+                        false,
+                        new SchemaShape(MetadataUtils.GetTrainerOutputMetadata()))
+            };
+        }
+
+        protected override BinaryPredictionTransformer<PcaPredictor> MakeTransformer(PcaPredictor model, ISchema trainSchema)
+            => new BinaryPredictionTransformer<PcaPredictor>(Host, model, trainSchema, _featureColumn);
```

Comment:
>a, _featu [](start = 84, length = 9)

@Zruty do we have a threeshold, to pass?

Reply:
we use default

---
In reply to: [219663367](https://github.com/dotnet/machinelearning/pull/996#discussion_r219663367) [](ancestors = 219663367)